lib-image/src/main/java/ai/kognition/pilecv4j/image/Utils.java

```diff
@@ -1128,13 +1128,30 @@ public static Size preserveAspectRatio(final Mat mat, final Size newSize) {
         return preserveAspectRatio(mat.size(), newSize);
     }
 
+    /**
+     *
+     * @param originalMatSize The Size of the original image
+     * @param newSize The desired Size of the new image
+     * @return The size of the new image, scaled to match the height or width of {@param newSize}, whichever requires the percent change in scale - unless
```

Comment:
This is much clearer. The original intention of this method is to match the height or width of {@param newSize} such that the image does not *exceed* those dimensions yet also preserves the size. This is how FFMpeg does it and what we use it for in Object Detection- any additional space required for a detection is filled with black.

Reply:
Done